lib/wallet/send_on_chain.dart

```diff
@@ -11,18 +17,83 @@ class SendOnChain extends StatefulWidget {
 }
 
 class _SendOnChainState extends State<SendOnChain> {
+  String address = "";
+  int amount = 0;
+
   @override
   void initState() {
     super.initState();
+
+    final bitcoinBalance = context.read<BitcoinBalance>();
+    amount = bitcoinBalance.amount.asSats;
   }
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: AppBar(
-        title: const Text('Send Bitcoin'),
-      ),
-      body: const SafeArea(child: Text("tbd")),
-    );
+        appBar: AppBar(
+          title: const Text('Send Bitcoin'),
+        ),
+        body: SafeArea(
+            child: Padding(
+                padding: const EdgeInsets.all(20.0),
+                child: Column(children: [
+                  Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
+                    const Text("Address", style: TextStyle(color: Colors.grey)),
+                    TextFormField(
+                      initialValue: "",
+                      keyboardType: TextInputType.text,
+                      onChanged: (text) {
+                        setState(() {
+                          address = text;
+                        });
+                      },
+                    ),
+                    const SizedBox(
+                      height: 10.0,
+                    ),
+                    const Text("Amount", style: TextStyle(color: Colors.grey)),
+                    TextFormField(
+                      initialValue: amount.toString(),
+                      keyboardType: TextInputType.number,
+                      decoration: const InputDecoration(
+                        border: UnderlineInputBorder(),
+                      ),
+                      inputFormatters: <TextInputFormatter>[FilteringTextInputFormatter.digitsOnly],
+                      onChanged: (text) {
+                        setState(() {
+                          amount = text != "" ? int.parse(text) : 0;
+                        });
+                      },
+                    )
+                  ]),
+                  Expanded(
+                    child: Container(
+                      margin: const EdgeInsets.fromLTRB(0, 0, 20, 30),
+                      child: Column(
+                        mainAxisAlignment: MainAxisAlignment.end,
+                        children: [
+                          Row(
+                            mainAxisAlignment: MainAxisAlignment.end,
+                            children: [
+                              ElevatedButton(
+                                  onPressed: () async {
+                                    FLog.info(
+                                        text:
+                                            "Sending " + amount.toString() + " sats to " + address);
+                                    await api.sendToAddress(address: address, amount: amount);
```

Comment:
🤔 Can this call fail? and if yes, shouldn't we add an error handling? at least showing the user something did not work?